### PR TITLE
Enable renaming inaccessible symbols.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/4904

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @dpoeschl

File: src/VisualStudio/Xaml/Impl/Features/InlineRename/XamlEditorInlineRenameService.cs [hunk 1]
```diff
@@ -66,14 +67,16 @@ public InlineRenameInfo(IXamlRenameInfoService renameService, Document document,
...
                     renameInStrings: optionSet.GetOption(RenameOptions.RenameInStrings),
                     renameInComments: optionSet.GetOption(RenameOptions.RenameInComments),
                     cancellationToken: cancellationToken).ConfigureAwait(false);
 
-                references.AddRange(renameLocations.Select(ds => new InlineRenameLocation(ds.Document, ds.TextSpan)));
+                references.AddRange(renameLocations.Select(
+                    ds => new InlineRenameLocation(ds.Document, ds.TextSpan)));
 
-                return new InlineRenameLocationSet(_renameInfo, _document.Project.Solution, references);
+                return new InlineRenameLocationSet(_renameInfo, _document.Project.Solution,
```
Comment (dpoeschl) [inline src/VisualStudio/Xaml/Impl/Features/InlineRename/XamlEditorInlineRenameService.cs]: Two args on the first line and one on the second?

File: src/Workspaces/Core/Portable/Rename/RenameLocation.cs [hunk 2]
```diff
@@ -69,5 +64,118 @@ public override int GetHashCode()
...
+                        // type or namespace.  This is usually a wildly broken situation.  i.e.
+                        // now due to hiding, something like a field/property is being referenced
+                        // in a type location.  It is highly likely that this should not be
+                        // renamed.
+                        return false;
+
+                    case CandidateReason.NotAnEvent:
+                    case CandidateReason.NotAWithEventsMember:
+                        // it's unlikely that someone would be referencing a non-event in an 
+                        // event context.  Likely this location should be included.
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Rename/RenameLocation.cs]: should *not* be included

File: src/Workspaces/Core/Portable/Rename/RenameLocation.cs [hunk 3]
```diff
@@ -69,5 +64,118 @@ public override int GetHashCode()
...
+                        return false;
+
+                    case CandidateReason.NotAnAttributeType:
+                        // It's feasible that someon was referencing some type in an attribute
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Rename/RenameLocation.cs]: someon => someone

File: src/Workspaces/Core/Portable/Rename/RenameLocation.cs [hunk 4]
```diff
@@ -69,5 +64,118 @@ public override int GetHashCode()
...
+                        // namespace, then it's likely something has gone very wrong (similar to
+                        // NotATypeOrNamespace), and we shouldn't update this reference.
+                        return false;
+
+                    case CandidateReason.NotInvocable:
+                        // Happens when something like a variable is being invoked, but the variable
+                        // isn't a delegate type.  This may be because the user intends to give 
+                        // this value a delegate type, but hasn't done so yet.  We should still allow
+                        // renaming this reference for now.
+                        return false;
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Rename/RenameLocation.cs]: return true;

Also, there should probably be tests that cover all of these scenarios.

File: src/Workspaces/Core/Portable/Rename/RenameLocation.cs [hunk 5]
```diff
@@ -69,5 +64,118 @@ public override int GetHashCode()
...
+                        // don't.  As overloading methods is very common, we won't update this location
+                        // as it might update code the user really doesn't want us to be touching.
+                        return false;
+
+                    case CandidateReason.LateBound:
+                        // This is a late bound call, so we should not update this location. 
+                        return false;
+
+                    case CandidateReason.Ambiguous:
+                        // we shoudl not touch amgiguous code.  We have no way to feel confident that
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Rename/RenameLocation.cs]: `We should` (capitalization [you were consistent everywhere else] & spelling), also `ambiguous` spelling

**Comment by CyrusNajmabadi** [comment]
Merging in as feedback was addressed.

**Pull request merged**